code/datums/gamemodes/gangwar.dm

```diff
@@ -630,30 +966,73 @@ proc/broadcast_to_all_gangs(var/message)
 	flags = FPRINT | EXTRADELAY | TABLEPASS | CONDUCT
 	w_class = W_CLASS_SMALL
 	var/in_use = 0
+	var/empty = FALSE
 
-	afterattack(target as turf|obj, mob/user as mob)
-		if(!istype(target, /turf) && !istype(target, /obj/decal/cleanable/gangtag))
-			return
 
-		if (!user)
-			return
-
-		if(in_use)
-			boutput(user, SPAN_ALERT("You are already tagging an area!"))
-			return
-
-		var/turf/turftarget = get_turf(target)
-
-		if((turftarget == loc) || (BOUNDS_DIST(src, target) > 0))
-			return
+	proc/check_tile_unclaimed(turf/target, mob/user)
+		for (var/obj/decal/gangtag/tag in orange(GANG_TAG_SIGHT_RANGE,target))
+			if(!IN_EUCLIDEAN_RANGE(tag, target, GANG_TAG_SIGHT_RANGE)) continue
+			if (tag.owners == user.get_gang())
+				boutput(user, "<span class='alert'>This is too close to an existing tag!</span>")
+				return
+		for (var/obj/ganglocker/locker in orange(GANG_TAG_SIGHT_RANGE_LOCKER,target))
+			if(!IN_EUCLIDEAN_RANGE(locker, target, GANG_TAG_SIGHT_RANGE_LOCKER)) continue
+			if (locker.gang == user.get_gang())
+				boutput(user, "<span class='alert'>This is too close to your locker!</span>")
+				return
 
-		var/datum/gang/gang = user.get_gang()
+		var/obj/decal/gangtag/existingTag
+		for (var/obj/decal/gangtag/turfTag in target.contents)
+			if (turfTag.active)
+				existingTag = turfTag
+
+		var/validLocation = FALSE
+		if (existingTag)
+			if (existingTag.owners != user.get_gang())
+				//if we're tagging over someone's tag, double our search radius
+				//(this will find any tags whose influence intersects with the target tag's influence)
+				for (var/obj/ganglocker/locker in orange(GANG_TAG_INFLUENCE_LOCKER+GANG_TAG_INFLUENCE,target))
+					if(!IN_EUCLIDEAN_RANGE(locker, target, GANG_TAG_INFLUENCE_LOCKER+GANG_TAG_INFLUENCE)) continue
+					if (locker.gang == user.get_gang())
+						validLocation = TRUE
+				for (var/obj/decal/gangtag/otherTag in orange(GANG_TAG_INFLUENCE*2,target))
+					if(!IN_EUCLIDEAN_RANGE(otherTag, target, GANG_TAG_INFLUENCE*2)) continue
+					if (otherTag.owners && otherTag.owners == user.get_gang())
+						validLocation = TRUE
+			else
+				boutput(user, "<span class='alert'>You can't spray over your own tags!</span>")
+				return
+		else
+			//we're tagging, check it's in our territory and not someone else's territory
+			for (var/obj/decal/gangtag/tag in orange(GANG_TAG_INFLUENCE,target))
+				if(!IN_EUCLIDEAN_RANGE(tag, target, GANG_TAG_INFLUENCE)) continue
+				if (tag.owners == user.get_gang())
+					validLocation = TRUE
+				else if (tag.owners)
+					boutput(user, "<span class='alert'>You can't spray in another gang's territory! Spray over their tag, instead!</span>")
+					if (user.GetComponent(/datum/component/tracker_hud))
+						return
+					var/datum/game_mode/gang/mode = ticker.mode
+					if (!istype(mode))
+						return
+					user.AddComponent(/datum/component/tracker_hud/gang, get_turf(tag))
+					SPAWN(3 SECONDS)
```

Comment:
Small bug with this, if it happens twice, with the second case occurring a bit after the first, it will end sooner than 3 seconds for the second case

Reply:
probably not too criminal but if there's a sane macro that doesnt look awful for this im down to fix it